Migrate more tests from plugin/pxr to core mayaUsd

mattyjams opened the pull request:
All of these tests exercise functionality that is in mayaUsd itself and not specific to the `pxrUsd` plugin or other code in `plugin/pxr`.

This is in the continuing effort to thin out the `plugin/pxr` code leaving behind only the assembly-related stuff that we intend to ultimately get rid of.

mattyjams commented on test/lib/mayaUsd/fileio/utils/testXformStack.py:
```diff
@@ -15,33 +15,32 @@
     @classmethod
     def tearDownClass(cls):
         standalone.uninitialize()
-    
```
You may want to exclude commit 8e39493fd99d192f130e9d360f30c4b29255c548 or otherwise ignore whitespace when looking through this PR, since it causes GitHub to identify `testXformStack.py` as a "large diff".

mattyjams commented:
Rebased to resolve conflicts from the merge of #1039.

kxl-adsk reviewed (approved):
Thank you for continuing this migration, Matt. Not many comments from my side but I stopped for a second on `testProxyShape`. We want this test to only validate features of the base proxy shape class and studio-specific features should be tested in separate test files. To communicate this intent, how about we rename this test to `testProxyShapeBase` and operate on proxyShapeBase node (i.e. `mayaUsdProxyShapeBase` instead of `mayaUsdProxyShape`)

mattyjams commented:
Thanks @kxl-adsk! Good catch. I just rebased the changes and addressed your proxy shape base note in https://github.com/Autodesk/maya-usd/pull/1042/commits/6420c884e5353e0d1d9f237dcbcc27852d8a4a51.

kxl-adsk reviewed (approved):
LGTM, thank you!

mattyjams commented:
Hopefully one last amendment to this: I separated out all of the `pxrHdImagingShape`-related stuff into its own dedicated test in https://github.com/Autodesk/maya-usd/pull/1042/commits/a8cb3708e52c05633fc700d250147167c3f82107. While `mayaUsdProxyShape` will automatically create a `pxrHdImagingShape` when the Viewport 2.0 render delegate is disabled, `mayaUsdProxyShapeBase` will **not** (and should not, since it is also the base type for the Animal Logic proxy shape).

kxl-adsk reviewed (approved):
LGTM!
